src/vcpkg/base/downloads.cpp

```diff
@@ -883,6 +883,43 @@ namespace vcpkg
         return true;
     }
 
+    bool azcopy_to_asset_cache(DiagnosticContext& context,
+                               StringView raw_url,
+                               const SanitizedUrl& sanitized_url,
+                               const Path& file)
+    {
+        auto azcopy_cmd = Command{"azcopy"};
```

Comment:
Note: We can't call `fetch_tool` naïvely here to get `azcopy` because this is happening on the background / binary cache submission thread.

@ras0219-msft points out that a potential way to reduce complexity here would be adding an explicit azcopy provider which moves the tool fetch up front like we do for the other binary caching backends e.g. awscli.